### PR TITLE
[4.0] Refactor com_languages 'override' view

### PR DESCRIPTION
**Pull request by C-Lodder**
Pull Request for Issue #17225

### Summary of Changes

- Rewrite the jQuery in vanilla JS
- Fix the spinner not showing
- Styling improvements

### CODE REVIEW

**Comment by brianteeman** [comment]
Shouldn't this pr also include changes to the unminified permission is file?

**Comment by C-Lodder** [comment]
@brianteeman reverted it just before your comment

File: media/com_languages/js/overrider.js [hunk 1]
```diff
@@ -67,90 +66,92 @@ Joomla.overrider.refreshCache = function()
...
-		if (r.data)
-		{
-			if (r.data.results)
-			{
-				self.insertResults(r.data.results);
+	Joomla.request({
+		url: 'index.php?option=com_languages&task=strings.search&format=json&searchstring=' 
+			+ self.states.searchstring + '&searchtype=' + self.states.searchtype + '&more=' + more,
+		method: 'POST',
+		method: 'POST',
```
Comment (dgrammatiko) [inline media/com_languages/js/overrider.js]: you can delete the duplicate here

**Review by dgrammatiko — CHANGES_REQUESTED** [review]

File: media/com_languages/js/overrider.js [hunk 2]
```diff
@@ -67,90 +66,92 @@ Joomla.overrider.refreshCache = function()
...
-		url: 'index.php?option=com_languages&task=strings.search&format=json',
-		data: 'searchstring=' + self.states.searchstring + '&searchtype=' + self.states.searchtype + '&more=' + more,
-		dataType: 'json'
-	}).done(function (r)
-	{
-		if (r.error && r.message)
-		{
-			alert(r.message);
+		var children = document.querySelectorAll('#results-container div.language-results');
+		for (var i = 0; children.lendth > 0; i++) {
```
Comment (dgrammatiko) [inline media/com_languages/js/overrider.js]: `for (var i = 0, l = children.lendth ; i < l; i++) {`

**Review by dgrammatiko — CHANGES_REQUESTED** [review]

File: media/com_languages/js/overrider.js (hunk 1 above)
Comment (C-Lodder) [reply media/com_languages/js/overrider.js]: done

File: media/com_languages/js/overrider.js (hunk 2 above)
Comment (C-Lodder) [reply media/com_languages/js/overrider.js]: and done

Comment (C-Lodder) [reply media/com_languages/js/overrider.js]: Glad you spotted this, as I misspelled `length`

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on cbc6325ede59dfcaeb2c707c6b316683ed14ce45<br /><br />First search works but any subsequent searches do not<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17334">issues.joomla.org/tracker/joomla-cms/17334</a>.</sub>

**Comment by brianteeman** [comment]
these movies show the issue

### before PR
before

### after pr
j4

File: media/com_languages/js/overrider.js [hunk 3]
```diff
@@ -67,90 +66,91 @@ Joomla.overrider.refreshCache = function()
...
+				}
+
+				if (response.data.more) {
+					// If there are more results than the sent ones
+					// display the more link
+					self.states.more = response.data.more;
+					moreResults.classList.add('show');
+				}
+				else {
+					moreResults.classList.add('show');
```
Comment (dgrammatiko) [inline media/com_languages/js/overrider.js]: we need to hide the results here

Comment (C-Lodder) [reply media/com_languages/js/overrider.js]: done

**Comment by C-Lodder** [comment]
@brianteeman - updated/fixed. Would you mind testing again please?

**Comment by brianteeman** [comment]
Will do after the f1

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 8f3cd72945089b869d13586c69bedb68ea2e9a0b<br /><br />all ok now<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17334">issues.joomla.org/tracker/joomla-cms/17334</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 8f3cd72945089b869d13586c69bedb68ea2e9a0b<br /><br />all ok now<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17334">issues.joomla.org/tracker/joomla-cms/17334</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 8f3cd72945089b869d13586c69bedb68ea2e9a0b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17334">issues.joomla.org/tracker/joomla-cms/17334</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

**Pull request merged**